rust/bin-shared/src/network_changes/windows.rs

```diff
@@ -685,3 +686,16 @@ mod async_dns {
         }
     }
 }
+
+/// Wraps an MPSC channel of capacity 1 to use as a cancel-safe notifier
```

Comment:
Can we encode this in the type-system by also creating the equivalent receiver here and statically construct it with a size of 1?